docs/internals.rst

```diff
@@ -29,45 +29,25 @@ The ``scenic`` module itself provides the top-level API for using Scenic: see :d
 How Scenic is Compiled
 ----------------------
 
-.. note::
-
-   This section describes the current compiler, which goes through various convolutions
-   in order to make use of the Python parser as much as possible. This has the advantage
-   of allowing almost all Python constructs to be used in Scenic, but the disadvantage of
-   sometimes producing cryptic error messages. We are in the process of writing a native
-   parser for Scenic which will replace Phases 1-4 below.
-
 The process of compiling a Scenic program into a `Scenario` object can be split into several phases.
 Understanding what each phase does is useful if you plan to modify the Scenic language.
 
-Phase 1: Import Handling
-~~~~~~~~~~~~~~~~~~~~~~~~
-In this phase the program is segmented into blocks ending with :keyword:`import` statements, which will then be compiled separately.
-This is done so that if a Scenic module defining new Scenic classes is imported, subsequent parsing can properly detect instantiations of those classes.
-For more details on the various rules for import statements in Scenic, see :keyword:`import`.
-
-Phase 2: Token Translation
-~~~~~~~~~~~~~~~~~~~~~~~~~~
-In this phase the Scenic program is tokenized using the Python tokenizer.
-Various Scenic constructs yield sequences of tokens which are meaningless to the Python parser, which we will use to parse the code in the next phase.
-To remedy this, all Scenic-specific token sequences are translated into new sequences that will pass through the Python parser if they are syntactically-valid Scenic.
+If you want to make a change to the language, also see :doc:`compiler`.
 
-Phase 3: Python Parser
+Phase 1: Scenic Parser
 ~~~~~~~~~~~~~~~~~~~~~~
-In this phase we parse the token stream resulting from Phase 2 using the Python parser.
-If the tokens do not parse properly, they come from a malformed Scenic program, and an error will be raised here.
-Unfortunately, since the tokens being parsed are not the original Scenic tokens and likely not semantically-valid Python either, the error messages can be confusing.
+In this phase the program is parsed using the Scenic parser. The parser is generated from a PEG grammar (``scenic.gram``) using `the Pegen parser generator <https://we-like-parsers.github.io/pegen/index.html>`_.
+The parser generates an abstract syntax tree (Scenic AST) for the program. Scenic AST is a superset of Python AST defined in ``ast.py`` and has additional nodes for representing Scenic-specific constructs.
```

Comment:
Maybe we should add a link here to the new guide, for people who want more details?